emcc.py

```diff
@@ -2074,6 +2074,14 @@ def default_setting(name, new_default):
   else:
     settings.JS_LIBRARIES.append((0, 'library_pthread_stub.js'))
 
+  if (
+      "$formatException" in settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE or
+      settings.INCLUDE_FULL_LIBRARY and not settings.DISABLE_EXCEPTION_CATCHING
+  ):
+    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$formatException']
+    settings.EXPORTED_FUNCTIONS += ['_emscripten_format_exception', '_free']
+    settings.EXPORTED_RUNTIME_METHODS += ['formatException']
```

Comment:
I would prefer not to modify the compiler like this if possible.. although I admit there is currently not good mechanism for adding JS -> native symbol dependencies.

We do have an open issue to improve that #15982.   So I'm inclined to remove this whole block for now to keep things simple .. or at least add a comment to remove it later.

BTW, I think you can add `formatException` to `EXPORTED_FUNCTIONS` on the line above and avoid the `EXPORTED_RUNTIME_METHODS` completely.

Reply:
If we remove this block entirely, `test_full_js_library_except` will fail.